src/librados/IoCtxImpl.cc

```diff
@@ -674,8 +684,18 @@ int librados::IoCtxImpl::operate(const object_t& oid, ::ObjectOperation *o,
 int librados::IoCtxImpl::operate_read(const object_t& oid,
 				      ::ObjectOperation *o,
 				      bufferlist *pbl,
-				      int flags)
-{
+				      int flags, optional_span* parent_span)
+{
+  #ifdef WITH_JAGER
+    Span span_1;
+    std::string span_name = "";
+    span_name = span_name = span_name+__FILENAME__+" function:"+__PRETTY_FUNCTION__;
+    if(parent_span)
+      trace(span_1, *parent_span, span_name.c_str());
+    optional_span this_parent_span(span_1);
+  #else
+    optional_span this_parent_span;
+  #endif
```

Comment:
"there shouldn't be any #ifdef WITH_JAGER in the code we're tracing" Noted.

"we shouldn't have to allocate std::strings to name the spans" the thing  I m trying to do is merge two things one is file name and another is function name so I need a string to hold those merge values because each of those macros returns const char*, using `string_view` is also not possible because I m my tracing functions demands a const char* so to convert a string_view I need to convert it into string, which  will be same as expensive as my current implmentation, should I removed the file name from the span name?
"optional_span doesn't appear to do anything; std::unique_ptr<Span> is already optional because it can be null", 
```
void func(optional_parameter=default_value)
```
if optional_parameter is `unique_ptr` then I wont be able to this because  `=` or something like this `Span&=nullptr` the second ones shows me nullptr, but if I wrap my unqiue_ptr inside of a struct then I could to do optional_parameter* = nullptr, this is why I m using optional_span as wrapper.

"trace() should return the Span instead of assigning it to the first argument" the the `Span` passed as first argument to trace is by reference so a scope of that `Span` is also alive inside the function it is created and it will get finished once we finish_trace() so void is okay isnt it?

Reply:
@cbodley will it be a matter of concern if while compiling my code WITH_JAGER=OFF I get a lot of petty warning likes use of unintialized varible?